internal/twirptest/client_test.go

```diff
@@ -282,6 +282,33 @@ func TestClientErrorsCanBeCaused(t *testing.T) {
 	}
 }
 
+func TestCustomHTTPClientInterface(t *testing.T) {
+	// Start up a server just so we can make a working client later.
+	h := PickyHatmaker(1)
+	s := httptest.NewServer(NewHaberdasherServer(h, nil))
+	defer s.Close()
+
+	// Create a custom wrapper to wrap our default client
+	httpClient := &wrappedHTTPClient{
+		Client: http.DefaultClient,
+	}
```

Comment:
This is a bit pedantic, but I want to be careful. We should have an assertion somewhere in here that we call the right `httpClient`.

Could you change `wrappedHTTPClient` to support a `wasCalled() bool` or something similar, and then check that we did call it? This test will obviously pass today, your code is quite clear, but I want to be protected from breaking it accidentally if we add more logic into the client constructor in the future.